src/com/haskforce/highlighting/annotation/HaskellDocumentationProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.haskforce.highlighting.annotation;
+
+import com.google.common.collect.Lists;
+import com.haskforce.highlighting.annotation.external.GhcMod;
+import com.haskforce.highlighting.annotation.external.GhcModi;
+import com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator;
+import com.haskforce.highlighting.annotation.external.TypeInfoUtil;
+import com.haskforce.settings.ToolKey;
+import com.haskforce.utils.ExecUtil;
+import com.intellij.lang.documentation.AbstractDocumentationProvider;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.LogicalPosition;
+import com.intellij.openapi.editor.VisualPosition;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtilCore;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class HaskellDocumentationProvider extends AbstractDocumentationProvider {
```

Comment:
Can you tell me in a little more detail what doesn't seem to work? I'll try to look into it as soon as I have beaten my desktop back into submission again, it doesn't want to build nor run the plugin anymore. My laptop still has it and runs so likely a config issue, as soon as I've got it sorted out I'll check it.